src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -1,37 +1,396 @@
+            this.loadingText = this.CUSTOM_LABELS.geTextUpdating;
+            inMemoryDataImport = this.prepareInMemoryDataImportForUpdate(inMemoryDataImport);
+        } else {
+            this.loadingText = this.CUSTOM_LABELS.geTextSaving;
+        }
+
+        inMemoryDataImport[PAYMENT_AUTHORIZE_TOKEN__C] = await this.tokenPromise;
+        this.dataImportRecord = await upsertDataImport({ dataImport: inMemoryDataImport });
+    }
+
+    /*******************************************************************************
+    * @description Re-apply Data Import id and relevant payment/elevate fields.
+    * The inMemoryDataImport is built new from the form on every save click. We
+    * need to catch it up with the correct id to make sure we update instead of
+    * inserting a new record on re-save attempts.
+    *
+    * @param {object} inMemoryDataImport: DataImport__c object built from the form
+    * fields.
+    *
+    * @return {object} inMemoryDataImport: DataImport__c object built from the form
+    * fields.
+    */
+    prepareInMemoryDataImportForUpdate(inMemoryDataImport) {
+        inMemoryDataImport.Id = this.dataImportRecord.Id;
+        inMemoryDataImport[PAYMENT_METHOD__C] = this.dataImportRecord[PAYMENT_METHOD__C];
+        inMemoryDataImport[PAYMENT_STATUS__C] = this.dataImportRecord[PAYMENT_STATUS__C];
+        inMemoryDataImport[PAYMENT_DECLINED_REASON__C] =
+            this.dataImportRecord[PAYMENT_DECLINED_REASON__C];
+
+        return inMemoryDataImport;
+    }
+    /*******************************************************************************
+    * @description Method attempts to make a purchase call to Payment
+    * Services. Immediately attempts to the charge the card provided in the Payment
+    * Services iframe (GE_TokenizeCard).
+    *
+    * @param {object} dataImportRecord: A DataImport__c record
+    */
+    processPayment = async () => {
+        this.loadingText = this.CUSTOM_LABELS.geTextChargingCard;
+
+        const isReadyToCharge = this.checkPaymentTransactionStatus(this.dataImportRecord[PAYMENT_STATUS__C]);
+        if (isReadyToCharge) {
+
+            const purchaseResponse = await this.makePurchaseCall();
+            if (purchaseResponse) {
+
+                this.dataImportRecord[PAYMENT_STATUS__C] = this.getPaymentStatus(purchaseResponse);
+                this.dataImportRecord[PAYMENT_DECLINED_REASON__C] =
+                    this.getPaymentDeclinedReason(purchaseResponse);
+
+                this.dataImportRecord = await upsertDataImport({ dataImport: this.dataImportRecord });
+
+                const isFailedPurchase = purchaseResponse.statusCode !== 201;
+                if (isFailedPurchase) {
+
+                    let errors = this.getFailedPurchaseMessage(purchaseResponse);
+                    this.errorCallback(errors);
+                    return;
                 }
-            )
+            }
+        }
+
+        this.processDataImport();
```

Comment:
Should this call to processDataImport() get moved outside of the purchase call, or do they need to be bundled for some reason? It maybe just reads a little cleaner that way, whereas the current if/else on hasPaymentToProcess makes it look like we are *either* processing payment or processing data import, when in fact we process the DI in both cases. BUT, if it complicates the flow to move the processDataImport() call out of this method, let's keep it here!